internal/core/api.rs

```diff
@@ -122,6 +114,10 @@ impl Default for CloseRequestResponse {
 }
 
 impl Window {
+    pub fn new(platform_window_weak: alloc::rc::Weak<dyn crate::window::PlatformWindow>) -> Self {
```

Comment:
there should be documentation,  probably with an example on how to use Rc::new_cyclic to get there.

I guess an alternative API would be to wrap the new_cyclic in this function and pass a function that construct a PlatformWindow. But we can go over that later maybe.

Reply:
Yes. Let's try this out a bit first and then see. I've created #1502 to go through all of the docs of the new APIs.